filesystem/src/main/java/com/nytimes/android/external/fs/FSReader.java

```diff
@@ -51,4 +53,20 @@ public void call(Emitter<BufferedSource> emitter) {
             }
         }, Emitter.BackpressureMode.NONE);
     }
+
+
+    public Observable<BufferedSource>  readAll(@Nonnull final String type) throws FileNotFoundException {
+        return Observable
+                .from(fileSystem.list(type))
```

Comment:
This should be deferred. Else when one doesn't immediately subscribe he will get a wrong result because `list` is called when he calls `readAll` and the actual read is performed when he subscribed (in the meantime there could be new items / old items deleted).